### PR TITLE
feat(onboarding): Reskin

### PR DESCRIPTION
**Pull request by evanpurkhiser**
This reskins the onboarding experience. There is mostly minimal functionality changes.

Of note are:

 - Steps are now isolated by themselves, it no longer displays previous steps, only the currently active step.
 - The 'Learn more' tab has been removed.
 - Invite team members has been removed and replaced with a link to open the modal.
 - It's no longer possible to "return to a specific step" as it was in the old implementation (you can still click the 'Back' button, or link in the top left)

The whole experience is best viewed using the development preview: Ensure you are added to the `sentry-test` org, then first:

Login: https://sentry-ayhdqsyhs.sentry.dev/auth/login/
Click here to see it in action: https://sentry-ayhdqsyhs.sentry.dev/onboarding/sentry-test/welcome/

image

image

image

### CODE REVIEW

File: src/sentry/static/sentry/app/stores/hookStore.tsx [hunk 1]
```diff
@@ -44,7 +44,6 @@ const validHookNames = new Set<HookName>([
   'member-invite-modal:customization',
   'metrics:event',
   'onboarding:extra-chrome',
-  'onboarding:invite-members',
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/stores/hookStore.tsx:47]: I'll be removing this from getsentry as well.

File: src/sentry/static/sentry/app/views/onboarding/components/firstEventIndicator.tsx [hunk 2]
```diff
@@ -7,43 +7,65 @@ import {IconCheckmark} from 'app/icons';
...
 type FirstIssue = null | true | Group;
 
-const FirstEventIndicator = (props: EventWaiterProps) => (
+type RenderProps = {
+  indicator: React.ReactNode;
+  firstEventButton: React.ReactNode;
+};
+
+type Props = EventWaiterProps & {
+  children: (props: RenderProps) => React.ReactNode;
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/views/onboarding/components/firstEventIndicator.tsx:23]: This has become a render-prop style component so we have more control over the layout. at the parent level

**Review by matejminar — COMMENTED** [review]
This is looking awesome.

File: src/sentry/static/sentry/app/views/onboarding/onboarding.tsx [hunk 3]
```diff
@@ -163,89 +152,102 @@ class Onboarding extends React.Component<Props, State> {
...
 
 const Container = styled('div')`
-  padding: 0 ${space(3)};
-  max-width: ${p => p.theme.breakpoints[0]};
+  display: flex;
+  justify-content: center;
+  position: relative;
+  background: ${p => p.theme.background};
+  padding: 120px ${space(3)};
+  padding-top: 160px;
```
[the remark this answers is not in the document]
Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/views/onboarding/onboarding.tsx:236]: Yeah I agree, maybe we can use a `vh` unit for this

**Comment by EvanPurkhiser** [comment]
Going to do some work here when we get back to handle some pre-loading for performance reasons

File: src/sentry/static/sentry/app/views/onboarding/welcome.tsx [hunk 4]
```diff
@@ -1,77 +1,130 @@
...
 
 type Props = StepProps & {
   organization: Organization;
-  config: Config;
 };
 
-const OnboardingWelcome = ({organization, onComplete, config, active}: Props) => {
-  const {user} = config;
+const easterEggText = [
+  t("Wan't to destroy real errors? Get started."),
```
Comment (dashed) [inline src/sentry/static/sentry/app/views/onboarding/welcome.tsx:24]:
```suggestion
  t("Want to destroy real errors? Get started."),
```

unsure if this is a typo

Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/views/onboarding/welcome.tsx:24]: LOL, it is but that's funny.

File: src/sentry/static/sentry/app/views/onboarding/onboarding.tsx [hunk 5]
```diff
@@ -163,89 +152,102 @@ class Onboarding extends React.Component<Props, State> {
...
+    );
   }
 
+  Contents = () => {
+    const cornerVariantControl = useAnimation();
+    const updateCornerVariant = () => {
+      cornerVariantControl.start(this.activeStepIndex === 0 ? 'top-right' : 'top-left');
+    };
+
+    // XXX(epurkhiser): We're using a react hook her becuase there's no other
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/onboarding/onboarding.tsx:183]:
```suggestion
    // XXX(epurkhiser): We're using a react hook here becuase there's no other
```

File: src/sentry/static/sentry/app/views/onboarding/projectSetup.tsx [hunk 6]
```diff
@@ -139,56 +139,96 @@ class ProjectDocs extends React.Component<Props, State> {
...
+        </TitleContainer>
+        <motion.p
+          variants={{
+            initial: {opacity: 0},
+            animate: {opacity: 1},
+            exit: {opacity: 0},
+          }}
+        >
+          {tct(
+            'Don’t have a relationship with your terminal? [link:Invite your team instead].',
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/onboarding/projectSetup.tsx:174]: Should this be an ascii quote?
```suggestion
            "Don't have a relationship with your terminal? [link:Invite your team instead].",
```

File: src/sentry/static/sentry/app/views/onboarding/welcome.tsx [hunk 7]
```diff
@@ -1,77 +1,130 @@
...
+            <PositionedFallingError>{fallingError}</PositionedFallingError>
+          </CTAContainer>
+          <SecondaryAction {...fadeAway}>
+            {tct('[flavorText][br][exitLink:Skip onboarding].', {
+              br: <br />,
+              exitLink: <Button priority="link" onClick={skipOnboarding} href="/" />,
+              flavorText:
+                fallCount > 0
+                  ? easterEggText[fallCount - 1]
+                  : t('Geez Mom, I’ve used Sentry before.'),
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/onboarding/welcome.tsx:78]: Should this be an ascii quote?
```suggestion
                  : t("Geez Mom, I've used Sentry before."),
```

**Comment by EvanPurkhiser** [comment]
I have some performance tweaks for preloading Paltform icons, but going to save those for a future PR since I have to merge some stuff in the platform icons repo first.

**Comment by markstory** [comment]
I got the following build error when trying to run this branch locally within getsentry:

```
ERROR in /Users/markstory/code/getsentry/static/getsentry/gsApp/components/onboardingInviteMembers.jsx
Module not found: Error: Can't resolve 'sentry/views/onboarding/projectSetup/inviteMembers' in '/Users/markstory/code/getsentry/static/getsentry/gsApp/components'
 @ /Users/markstory/code/getsentry/static/getsentry/gsApp/components/onboardingInviteMembers.jsx 28:0-79 152:29-42 229:149-162
 @ /Users/markstory/code/getsentry/static/getsentry/gsApp/registerHooks.tsx
 @ /Users/markstory/code/getsentry/static/getsentry/gsApp/index.tsx
```

**Comment by EvanPurkhiser** [comment]
Oops, sorry mark, you'll have to pull in this PR too to test. https://github.com/getsentry/getsentry/pull/4889

File: src/sentry/static/sentry/app/views/onboarding/onboarding.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/views/onboarding/onboarding.tsx]: Adjusted, should feel better now

File: src/sentry/static/sentry/app/stores/hookStore.tsx (hunk 1 above)
Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/stores/hookStore.tsx:47]: Removed here https://github.com/getsentry/getsentry/pull/4889

File: src/sentry/static/sentry/app/views/onboarding/platform.tsx [hunk 8]
```diff
@@ -118,57 +110,66 @@ class OnboardingPlatform extends React.Component<Props, State> {
...
+          <p>
+            {tct(
+              `Don't see your platform-of-choice? Fear not. Select
+               [otherPlatformLink:other platform] to use a [communityClient:community client].
+               Need help? Learn more in [docs:our docs].`,
+              {
+                otherPlatformLink: (
+                  <Button
+                    priority="link"
+                    onClick={() => this.handleSetPlatform('other')}
```
Comment (dashed) [inline src/sentry/static/sentry/app/views/onboarding/platform.tsx:151]: I think when this button is clicked, it should add `platform="other"` prop to the `<PlatformPicker />` component above.


https://user-images.githubusercontent.com/139499/103940262-0f619980-50fb-11eb-9d37-30258ddccff4.mp4

Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/views/onboarding/platform.tsx:151]: Yeah I was trying to workout a decent fix for this, I do agree.

File: src/sentry/static/sentry/app/views/onboarding/platform.tsx [hunk 9]
```diff
@@ -118,57 +110,66 @@ class OnboardingPlatform extends React.Component<Props, State> {
...
+          </p>
+          <PlatformPicker
+            noAutoFilter
+            platform={selectedPlatform}
+            setPlatform={this.handleSetPlatform}
+          />
+          <p>
+            {tct(
+              `Don't see your platform-of-choice? Fear not. Select
+               [otherPlatformLink:other platform] to use a [communityClient:community client].
```
Comment (dashed) [inline src/sentry/static/sentry/app/views/onboarding/platform.tsx:145]: I think this should be reworded to be: Select *other platform* when using a *community client*.

**Review by dashed — APPROVED** [review]
Other than the minor nitpicks. I think the reskin looks good to me 👍

**Pull request merged**